versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/VersionStoreImpl.java

```diff
@@ -494,8 +496,8 @@ public PaginationIterator<KeyEntry> getKeys(Ref ref, String pagingToken, boolean
     StoreKey begin =
         pagingToken != null && !pagingToken.isEmpty()
             ? keyFromString(fromString(pagingToken).token().toStringUtf8())
-            : null;
-    StoreKey end = null;
+            : (minKey != null ? keyToStoreKey(minKey) : null);
+    StoreKey end = maxKey != null ? keyToStoreKey(maxKey) : null;
```

Comment:
Will this work when paginating over a "prefix" query? I mean that on the second page min and max keys will no longer be equals when passed to `index.iterator(...)` :thinking:

Reply:
The max key will still be the same, and "min" would be "replaced" with the key from the paging token.

Reply:
that's exactly what I mean. If we start with min == max for a prefix query, min != max in the second page... Just wondering if we want this to work for prefix queries as well as for normal ranges..

Reply:
Sorry, my comment is not relevant to the latest code - resolving.